diff -Nur orig/chanman.cc mod/chanman.cc
--- orig/chanman.cc	2012-07-09 17:02:58.000000000 +0200
+++ mod/chanman.cc	2018-05-03 08:17:11.711421076 +0200
@@ -54,9 +54,10 @@
 ;
 
 void cMyChannels::Set() {
+    LOCK_CHANNELS_READ;
 	int pos = 1;
-	for (cChannel *channel = Channels.First(); channel;
-			channel = Channels.Next(channel)) {
+    for (const cChannel *channel = Channels->First(); channel;
+			channel = Channels->Next(channel)) {
 		Add(new cMyChannel(channel, pos));
 		pos++;
 	}
@@ -327,10 +328,10 @@
 	int positem = 0;
 	number = 0;
 	bool isInGroup = false;
-
+	LOCK_CHANNELS_READ;
 	cMyChannel *first = NULL;
-	cChannel *currentChannel;
-	currentChannel = Channels.GetByNumber(cDevice::CurrentChannel());
+	const cChannel *currentChannel;
+	currentChannel = Channels->GetByNumber(cDevice::CurrentChannel());
 
 	cItemChoice1 *currentItem = NULL;
 
@@ -700,11 +701,12 @@
 
 cSecondMenu::cSecondMenu(cMyChannel *firstlch, int numch) :
 		cOsdMenu(tr("Channels"), 4, 25) {
+	LOCK_CHANNELS_READ;
 	cMyChannel *lchannel;
 	number = 0;
 	lchannel = firstlch;
-	cChannel *currentChannel;
-	currentChannel = Channels.GetByNumber(cDevice::CurrentChannel());
+	const cChannel *currentChannel;
+	currentChannel = Channels->GetByNumber(cDevice::CurrentChannel());
 	cItemChoice2 *currentItem = NULL;
 
 	for (int i = 1; i <= numch; i++) {
@@ -746,7 +748,8 @@
 
 void cSecondMenu::Propagate(void) {
 	int i = 1;
-	Channels.ReNumber();
+	LOCK_CHANNELS_WRITE;
+	Channels->ReNumber();
 	for (cItemChoice2 *ci = (cItemChoice2 *) First(); ci; ci =
 			(cItemChoice2 *) ci->Next()) {
 		ci->SetPos(i);
@@ -754,22 +757,22 @@
 		ci->Set();
 	}
 	Display();
-	Channels.SetModified(true);
+	Channels->SetModified();
 	MyChannels.SetModified();
 }
 
 void cSecondMenu::Move(int From, int To) {
+	LOCK_CHANNELS_WRITE;
 	int CurrentChannelNr = cDevice::CurrentChannel();
-	cChannel *CurrentChannel = Channels.GetByNumber(CurrentChannelNr);
+	cChannel *CurrentChannel = Channels->GetByNumber(CurrentChannelNr);
 	cChannel *FromChannel = GetChannel(From); // From e' la posizione nel menu di cItemChoice2
 	cChannel *ToChannel = GetChannel(To); // GetChannel Restituisce la posizione nella lista Channels.
 	cMyChannel *FromMyChannel = GetMyChannel(From);
 	cMyChannel *ToMyChannel = GetMyChannel(To);
-
 	if (FromChannel && ToChannel) {
 		int FromNumber = FromChannel->Number();
 		int ToNumber = ToChannel->Number();
-		Channels.Move(FromChannel, ToChannel);
+		Channels->Move(FromChannel, ToChannel);
 		MyChannels.Move(FromMyChannel, ToMyChannel);
 		cOsdMenu::Move(From, To);
 		Propagate();
@@ -778,7 +781,7 @@
 		if (CurrentChannel && CurrentChannel->Number() != CurrentChannelNr) {
 			if (!cDevice::PrimaryDevice()->Replaying()
 					|| cDevice::PrimaryDevice()->Transferring())
-				Channels.SwitchTo(CurrentChannel->Number());
+				Channels->SwitchTo(CurrentChannel->Number());
 			else
 				cDevice::SetCurrentChannel(CurrentChannel);
 		}
@@ -790,14 +793,15 @@
 	//  int CurrentChannelNr = cDevice::CurrentChannel();
 	//  cChannel *CurrentChannel = Channels.GetByNumber(CurrentChannelNr);
 
+	LOCK_CHANNELS_WRITE;
 	cMyChannel *FromMyChannel;
 
 	cItemChoice2 *CurrentChoice = (cItemChoice2 *) Get(Current());
 
 	cMyChannel *ToMyChannel = CurrentChoice->GetMyChan();
 
-	cChannel *FromChannel;
-	cChannel *ToChannel = ToMyChannel->GetCh();
+	const cChannel *FromChannel;
+	const cChannel *ToChannel = ToMyChannel->GetCh();
 
 	int i = 1;
 	for (cToCutChannel *mychancut = ToCutChannels.First(); mychancut;
@@ -808,7 +812,10 @@
 
 		FromChannel = FromMyChannel->GetCh();
 
-		Channels.Move(FromChannel, ToChannel);
+        int FromNumber = FromChannel->Number();
+ 		int ToNumber = ToChannel->Number();
+            
+        Channels->Move(FromNumber, ToNumber);
 		MyChannels.Move(FromMyChannel, ToMyChannel);
 
 		//void cOsdMenu::Add(cOsdItem *Item, bool Current, cOsdItem *After)
@@ -855,7 +862,7 @@
 }
 
 eOSState cSecondMenu::Switch(void) {
-	cChannel *ch = ((cItemChoice2 *) Get(Current()))->GetChan();
+	const cChannel *ch = ((cItemChoice2 *) Get(Current()))->GetChan();
 	if (ch)
 		return cDevice::PrimaryDevice()->SwitchChannel(ch, true) ?
 				osEnd : osContinue;
@@ -996,7 +1003,7 @@
 		return 'C';
 }
 
-cItemChoice2::cItemChoice2(int Positem, cChannel *Channel,
+cItemChoice2::cItemChoice2(int Positem, const cChannel *Channel,
 		cMyChannel *MyChannel) {
 	positem = Positem;
 	channel = Channel;
diff -Nur orig/chanman.h mod/chanman.h
--- orig/chanman.h	2012-07-09 17:02:58.000000000 +0200
+++ mod/chanman.h	2018-05-02 19:19:16.695913845 +0200
@@ -39,13 +39,13 @@
  private:
   static eChannelSortMode sortMode;
   static eChannelFilterMode filterMode;
-  cChannel *channel;
+  const cChannel *channel;
   int pos;
   bool isIn;
   bool toCut;
  public:
-  cMyChannel(cChannel *Channel, int Pos) { channel = Channel; isIn=true;pos=Pos;toCut=false;}
-  cChannel *GetCh(void){return channel;}
+  cMyChannel(const cChannel *Channel, int Pos) { channel = Channel; isIn=true;pos=Pos;toCut=false;}
+  const cChannel *GetCh(void){return channel;}
   void SetCh(cChannel *Channel){ channel = Channel; isIn=true; }
   static void SetSortMode(eChannelSortMode SortMode){ sortMode=SortMode;}
   static void SetFilterMode(eChannelFilterMode FilterMode){ filterMode=FilterMode;}
@@ -160,11 +160,11 @@
  private:
   int positem;
   static bool increasing;
-  cChannel *channel;
+  const cChannel *channel;
   cMyChannel *mychannel;
  public:
-  cItemChoice2(int positem,cChannel *channel,cMyChannel *mychannel);
-  cChannel *GetChan(void){return channel;}
+  cItemChoice2(int positem, const cChannel *channel,cMyChannel *mychannel);
+  const cChannel *GetChan(void){return channel;}
   cMyChannel *GetMyChan(void){return mychannel;}
   int Number(void){return positem;}
   void SetPos(int Positem){positem=Positem;}
