diff -Nur orig/CMakeLists.txt mod/CMakeLists.txt
--- orig/CMakeLists.txt	2021-05-25 08:25:11.000000000 +0200
+++ mod/CMakeLists.txt	2021-05-25 08:29:26.042236349 +0200
@@ -130,7 +130,7 @@
 #----------------------- global options ------------------------------
 
 if (OSCamOperatingSystem MATCHES "Linux")
-	set (DEFAULT_CS_CONFDIR "/usr/local/etc")
+	set (DEFAULT_CS_CONFDIR "/etc/oscam")
 elseif (OSCamOperatingSystem MATCHES "Mac OS X")
 	if (NOT HAVE_PCSC EQUAL 0)
 		add_definitions ("-DWITH_PCSC=1")
